cartography/intel/aws/rds.py

```diff
@@ -98,6 +105,132 @@ def load_rds_instances(neo4j_session, data, region, current_aws_account_id, aws_
+        db_sng = instance.get('DBSubnetGroup')
+        neo4j_session.run(
+            attach_rds_to_subnet_group,
+            DBSubnetGroupName=db_sng.get('DBSubnetGroupName'),
+            VpcId=db_sng.get("VpcId", None),
+            DBSubnetGroupDescription=db_sng.get('DBSubnetGroupDescription', None),
+            DBSubnetGroupStatus=db_sng.get('SubnetGroupStatus', None),
+            DBInstanceArn=instance.get('DBInstanceArn'),
+            aws_update_tag=aws_update_tag
+        )
+        _attach_ec2_subnets_to_subnetgroup(neo4j_session, db_sng, aws_update_tag)
+
+
+def _attach_ec2_subnets_to_subnetgroup(neo4j_session, db_subnet_group, aws_update_tag):
+    """
+    Attach EC2Subnets to the DB Subnet Group.
+
+    From https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_VPC.WorkingWithRDSInstanceinaVPC.html:
+    `Each DB subnet group should have subnets in at least two Availability Zones in a given region. When creating a DB
+    instance in a VPC, you must select a DB subnet group. Amazon RDS uses that DB subnet group and your preferred
+    Availability Zone to select a subnet and an IP address within that subnet to associate with your DB instance.`
+    """
+    attach_subnets_to_sng = """
+    MATCH(subnet:EC2Subnet{subnetid:{SubnetIdentifier}}),
+    (sng:DBSubnetGroup{id:{DBSubnetGroupName}})
+    MERGE(sng)-[r:RESOURCE]->(subnet)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {aws_update_tag},
+    subnet.availability_zone = {SubnetAvailabilityZone}
+    """
+    # TODO: Note: RDS data has subnet availability zone data, but I don't think that this is the right place to
+    # TODO: update it.  We should update the availability zone data in the EC2 ingestion part.
+    if not db_subnet_group.get('DBSubnetGroupName'):
+        logger.debug("Expected DBSubnetGroup to have a DBSubnetGroupName but it doesn't.  Here is the object: %r",
+                     db_subnet_group)
+    else:
+        for sn in db_subnet_group.get('Subnets', []):
+            subnet_id = sn.get('SubnetIdentifier', None)
+            if not subnet_id:
+                logger.debug("Expected Subnet to have a SubnetIdentifier but it doesn't. Here is the object: %r",
+                             db_subnet_group)
+            else:
+                neo4j_session.run(
+                    attach_subnets_to_sng,
+                    SubnetIdentifier=subnet_id,
+                    DBSubnetGroupName=db_subnet_group.get('DBSubnetGroupName'),
+                    aws_update_tag=aws_update_tag,
+                    SubnetAvailabilityZone=sn.get('SubnetAvailabilityZone', {}).get('Name', None)
+                )
+
+
+def _attach_ec2_security_groups(neo4j_session, instance, aws_update_tag):
+    """
+    Attach an RDS instance to its EC2SecurityGroups
+    """
+    attach_rds_to_group = """
+    MATCH (rds:RDSInstance{id:{RdsArn}}),
+    (sg:EC2SecurityGroup{groupid:{GroupId}})
+    MERGE (rds)-[m:MEMBER_OF_EC2_SECURITY_GROUP]->(sg)
```

Comment:
BLOCK - missing cleanup

Reply:
The DBSubnetGroups are cleaned up [here](https://github.com/lyft/cartography/pull/10/files#diff-db18c81378ce7f90e09bd4f343906bedR4) and the RDSInstances are cleaned up [here](https://github.com/lyft/cartography/pull/10/files#diff-db18c81378ce7f90e09bd4f343906bedR9).

Reply:
What about the MEMBER_OF_EC2_SECURITY_GROUP relationship? modules doing MERGE should cleanup on their own and not assume other modules does it (ordering is not guaranteed)